src/WebJobs.Script.WebHost/Controllers/KeysController.cs

```diff
@@ -28,17 +29,38 @@ public class KeysController : Controller
         private readonly ILogger _logger;
         private readonly IOptions<ScriptApplicationHostOptions> _applicationOptions;
         private readonly IFileSystem _fileSystem;
+        private readonly IMetricsLogger _metrics;
+        private readonly string _metricsDescription;
 
-        public KeysController(IOptions<ScriptApplicationHostOptions> applicationOptions, ISecretManagerProvider secretManagerProvider, ILoggerFactory loggerFactory, IFileSystem fileSystem)
+        public KeysController(IOptions<ScriptApplicationHostOptions> applicationOptions, ISecretManagerProvider secretManagerProvider, ILoggerFactory loggerFactory, IFileSystem fileSystem, IMetricsLogger metrics)
```

Comment:
I don't really like the requirement that to use `[MetricsAction]` you also need to implement `IMetricsController` (and have `IMetricsLogger` injected).

The `IMetricsLogger` requirement can be removed if you do something similar to the `RequiresRunningHostAttribute` -- where you can inject services directly to the attribute and the controller doesn't have to know about it. I'd recommend doing that. 

That leaves us with the `MetricsDescription`... trying to think of a way to remove that requirement.

Reply:
Agreed. We already have a pattern established and that would be a cleaner approach for those cross cutting features. For the description, though it my be repetitive, to reduce the coupling, I would rather see that passed into the attribute as an argument.

Reply:
In this case, he needs the description to be dynamic, based on the repository type being used...

Reply:
I noticed that, but not sure what the exact requirements are. Do we need this to be repository metrics? If so, I'd question adding this to the controller to begin with.

Reply:
As we do not have good way to pass `MetricsDescription` into the attribute I sent out an another PR  that adds metrics directly in `SecretManager ` https://github.com/Azure/azure-functions-host/pull/4122